### DISCUSSION TITLE
How does Yolox assign FPN levels to ground-truth boxes?

### DISCUSSION THREAD

**Discussion by andrew-quaisley** [Megvii-BaseDetection/YOLOX#1479] [Q&A]
In the arXiv paper, it states: "We assign the center location of each object as the positive sample and pre-define a scale range, as done in [29], to designate the FPN level for each object."  [29] is the FCOS paper here: https://arxiv.org/pdf/1904.01355.pdf.  When I read this, I understood it to mean that ground-truth boxes are assigned an FPN level based on their size, and a prediction will only be matched to the ground-truth box if it comes from the corresponding FPN level.  However, I don't see where this is happening in the code!  Is this actually happening, or have I misunderstood something?  I want to know how the boxes get assigned to FPN levels for a semi-supervised version of Yolox I'm working on.  Can anyone help?  Thanks!

**Comment by gouldberg** [comment]
I just add info.
Yolox head gets output from FPN and convert to 8400 dim feature in head forward. (8400 = 80 * 80 + 40 * 40 + 20 * 20)
Image size is 640 * 640 --> 80 * 80 means each feature corresponds to each 8 * 8 image grid in 640 * 640.  (640 * 640 = 8 * 80 * 8 * 80)
40 * 40 means each feature corresponds to each 16 * 16 image grid in 640 * 640 (640 * 640 = 16 * 40 * 16 * 40), and so on.
